### ISSUE TITLE
TTNN Generic port `KernelsToCpp.cpp` to TTNN

### ISSUE THREAD

**Issue by nsmithtt** [tenstorrent/tt-mlir#203]
See #201 for details.

Currently `KernelsToCpp.cpp` lives as a TTMetal dialect transform, but perhaps we should refactor the parts of the code that make sense to the TTKernel area so that it could be mostly shared between the TTMetal and TTNN dialects.

What we want is the equivalent of `emitDispatchOpRegionAsCpp`, like `emitTTNNGenericOpRegionAsCpp` or something.

Related issue, to make `KernelsToCpp.cpp` a proper dialect conversion, once the below issue is Done `KernelsToCpp.cpp` will have moved to `lib/Conversion/TTKernelToEmitC`:
- https://github.com/tenstorrent/tt-mlir/issues/317

**Comment by staylorTT** [comment]
@nsmithtt Can we assign this to a new owner?

**Comment by nsmithtt** [comment]
This was actually already done
